docs/notebooks/demo/gretel-demo-sequence-of-events.ipynb

```diff
@@ -0,0 +1,456 @@
+    "- **Client Feedback Session**: Gathering feedback from clients or stakeholders.\n",
+    "- **Adjustments Based on Feedback**: Making changes to the project based on received feedback.\n",
+    "- **Deployment**: Release of the final product to the end-users or stakeholders.\n",
+    "- **Project Retrospective**: Reflective meeting to discuss what went well and what could be improved.\n",
+    "\n",
+    "### Workflow Logic:\n",
+    "The workflow is designed with a logical progression from initiation to closure, with mandatory events ensuring the project's essential milestones are met. Optional events provide opportunities to enhance project outcomes, address unforeseen challenges, or incorporate stakeholder feedback. Solid lines represent the flow between mandatory events, while dashed lines indicate where optional events can be integrated into the project lifecycle.\n",
+    "\n",
+    "This dataset and workflow visualization offer a comprehensive overview of the structured approach to managing projects, highlighting the flexibility to adapt to project-specific requirements and changes throughout the project"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# show data\n",
+    "from IPython.display import Image\n",
+    "\n",
+    "# Display an image with its filename\n",
+    "Image(filename=\"project_management_workflow.png\", width=400, height=200)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load and preview training data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "df = pd.read_csv(\n",
+    "    \"https://gretel-datasets.s3.us-west-2.amazonaws.com/project_management_sequences.csv\"\n",
+    ")\n",
+    "\n",
+    "EXAMPLE_COLUMN = \"PROJECT_ID\"\n",
+    "EVENT_COLUMN = \"EVENT_TYPE\"\n",
+    "\n",
+    "selected_projects = np.random.choice(\n",
+    "    df[EXAMPLE_COLUMN].unique(), 3, replace=False\n",
+    ")\n",
+    "\n",
+    "pd.set_option(\"display.max_rows\", None)\n",
+    "display(df[df[EXAMPLE_COLUMN].isin(selected_projects)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare data for DGAN"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from demo_utils import pad_sequence\n",
+    "\n",
+    "# prepare for DGAN\n",
+    "\n",
+    "max_len = df.groupby(EXAMPLE_COLUMN).size().max()\n",
+    "\n",
+    "# Pad each group and concatenate back into a DataFrame\n",
+    "data_source = pd.concat(\n",
+    "    [pad_sequence(group, max_len, example_id_column=EXAMPLE_COLUMN, event_column=EVENT_COLUMN) for _, group in df.groupby(EXAMPLE_COLUMN)],\n",
+    "    ignore_index=True,\n",
```

Comment:
Might be helpful to add `pad_value="[END]"` as an explicit keyword argument to `pad_sequence` here to help the user understand where the `[END]` value came from.